### PR TITLE
Rename ArchComponents to AndroidLifecycle.

### LINKED ISSUE #91: Rename ArchComponents stuff to AndroidLifecycle
A little simpler. We can add an overload for views too out of convenience.

### PR DESCRIPTION
**Pull request by lsvijay**
Rename ArchComponents to AndroidLifecycle.
Summary:
* Rename com.uber.autodispose.android.archcomponents package to com.uber.autodispose.android.lifecycle
* Rename AndroidLifecycleScopeProvider to just AndroidLifecycle

Resolves #91

### CODE REVIEW

**Comment by hzsweers** [comment]
Thanks!

**Comment by hzsweers** [comment]
Soooo after looking at this more and seeing it next to other usages, I think we should actually keep the `ScopeProvider`. It is a bit wordy, but it's become a pattern now. It does make it clear what it's for too

**Comment by lsvijay** [comment]
Agree, sounds good to me.

On Oct 1, 2017, at 12:50 AM, Zac Sweers <notifications@github.com<mailto:notifications@github.com>> wrote:


Soooo after looking at this more and seeing it next to other usages, I think we should actually keep the ScopeProvider. It is a bit wordy, but it's become a pattern now. It does make it clear what it's for too
